bin/oxstore

```diff
@@ -0,0 +1,304 @@
+    "Run the cat action"
+    for a_key in generate_log_file_s3_keys(
+            bucket,
+            type_name,
+            start_date,
+            end_date=end_date):
+        if not a_key.exists():
+            continue
+        elif a_key.name.endswith(".bz2"):
+            decompressor = bz2.BZ2Decompressor()
+            for data in a_key:
+                sys.stdout.write(decompressor.decompress(data))
+        else:
+            a_key.get_file(fp=sys.stdout)
+
+
+def do_download(bucket, type_name, start_date, end_date=None):
+    "Run the download action"
+    for key in generate_log_file_s3_keys(
+            bucket,
+            type_name,
+            start_date,
+            end_date=end_date):
+        if not key.exists():
+            continue
+        filename = os.path.basename(key.name)
+        log.info("Downloading %s", filename)
+        key.get_contents_to_filename(filename)
+
+
+def parse_date_range_arguments(parser, args):
+    try:
+        start_date = datetime.datetime.strptime(args.start, '%Y%m%d')
+    except ValueError:
+        parser.error("Bad format for date. Try YYYYMMDD")
+    end_date = None
+    if args.end:
+        try:
+            end_date = datetime.datetime.strptime(args.end, '%Y%m%d')
+        except ValueError:
+            parser.error("Bad format for date. Try YYYYMMDD")
+
+    return start_date, end_date
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Manage blueox logs')
+    parser.add_argument('--verbose', '-v', action='count')
+    parser.add_argument(
+        '--bucket', '-b', action='store', required=True)
+    parser.add_argument(
+        '--log-path', '-p', action='store', default=DEFAULT_LOG_PATH)
+
+    subparsers = parser.add_subparsers(
+        dest='command', help='available commands')
+    parser_archive = subparsers.add_parser(
+        'archive', help='Archive local logs')
+    parser_archive.add_argument(
+        '--no-remove',
+        action='store_true',
+        default=False,
+        help="Upload log files, but don't remove them locally")
+
+    parser_cat = subparsers.add_parser(
+        'cat', help='output specified log to stdout')
+    parser_cat.add_argument(
+        'log_type', action='store', nargs=1, help="log type")
+    parser_cat.add_argument(
+        '--start', '-s',
+        action='store', required=True, help="start date (YYYYMMDD)")
+    parser_cat.add_argument(
+        '--end', '-e', action='store', required=False, help="end date")
+
+    parser_dnld = subparsers.add_parser(
+        'download', help='download specified log file(s)')
+    parser_dnld.add_argument(
+        'log_type', action='store', nargs=1, help="log type")
+    parser_dnld.add_argument(
+        '--start', '-s',
+        action='store', required=True, help="start date (YYYYMMDD)")
+    parser_dnld.add_argument(
+        '--end', '-e', action='store', required=False, help="end date")
+
+    args = parser.parse_args()
+
+    if boto is None:
```

Comment:
I want to make this script work if you don't have boto installed. Like you can use all the none-s3 features. But that's obviously not yet how it's structured.

You'll notice I only added boto to requirements.txt, not to the setup.py dependencies.